test/e2e/apps/statefulset.go

```diff
@@ -510,6 +510,152 @@ var _ = SIGDescribe("StatefulSet", func() {
 
 		})
 
+		ginkgo.It("should perform canary updates and phased rolling updates of template modifications for partiton1 and delete pod-0", func(ctx context.Context) {
+			ginkgo.By("Creating a new StatefulSet")
+			ss := e2estatefulset.NewStatefulSet("ss2", ns, headlessSvcName, 3, nil, nil, labels)
+			setHTTPProbe(ss)
+			ss.Spec.UpdateStrategy = appsv1.StatefulSetUpdateStrategy{
+				Type: appsv1.RollingUpdateStatefulSetStrategyType,
+				RollingUpdate: func() *appsv1.RollingUpdateStatefulSetStrategy {
+					return &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32(1),
+					}
+				}(),
+			}
+			readyNode, err := e2enode.GetRandomReadySchedulableNode(ctx, c)
+			framework.ExpectNoError(err)
+			hostLabel := "kubernetes.io/hostname"
+			hostLabelVal := readyNode.Labels[hostLabel]
+			ss.Spec.Template.Spec.NodeSelector = map[string]string{hostLabel: hostLabelVal}
```

Comment:
Is this needed? If no, I would suggest to simplify.

Reply:
It is not needed. The reproduction rate of this issue was lower when there were multiple nodes so I thought forcing pods to schedule on the same node might make the test fail more consistently, but it didn't work, I will remove it.

Reply:
its removed as suggested